src/frontend/src/expr/type_inference/func.rs

```diff
@@ -157,6 +164,54 @@ fn infer_type_for_special(
                 _ => Ok(None),
             }
         }
+        ExprType::Equal
+        | ExprType::NotEqual
+        | ExprType::LessThan
+        | ExprType::LessThanOrEqual
+        | ExprType::GreaterThan
+        | ExprType::GreaterThanOrEqual
+        | ExprType::IsDistinctFrom
+        | ExprType::IsNotDistinctFrom => {
+            ensure_arity!("cmp", | inputs | == 2);
+            match (inputs[0].is_unknown(), inputs[1].is_unknown()) {
+                // `'a' = null` handled by general rules later
```

Comment:
Can this also be `null = null`?

Reply:
Yes. Both literal `null` and literals with single quotes (e.g. `'a'` or `'1'`) are considered to be `unknown` type.